[feat] domain 에 엔티티 클래스 추가

Highjune opened the pull request:
- Entity 추가
- DdoEatLog, Follow, LikeIt, User, Bookmark , Category

### ✅ PR 타입 & 이슈번호
ex) 기능 추가 (#8)

### 📌 상세 내용
- entity 추가함
- (기존 존재하는 Store 제외) DdoEatLog, Follow, LikeIt, User, Bookmark , Category
- 아직 패키지 정리가 되지 않은 것 같아서 domain 밑에 그냥 패키지 하나씩 만들어서 다 넣었습니다. 추후에 패키지 정리되면 위치는 옮겨야 할 것 같습니다.
- Like 테이블은 예약어 때문인지 ddl(true) 옵션으로 자동생성이 되지 않더군요. 그래서 LikeIt 으로 만들었습니다.
- 저희 프로젝트에서는 제대로 생성되는지 아직 확인은 안했으나, 해당 엔티티 코드 그대로 로컬에서 새로운 저장소 파서 mysql 에 테스트 해보니 잘 생성이 되는 것을 확인했습니다.

### 🔥 궁금한점
- Entity의 primitive 타입은 Wrapper 가 맞겠죠? null 이 들어올 수 있으니.

Highjune committed:
[feat] domain
- Entity 추가
- DdoEatLog, Follow, LikeIt, User, Bookmark , Category

Ryuhyis commented on server-domain/src/main/java/com/depromeet/entity/log/DdoEatLog.java:
```diff
@@ -0,0 +1,41 @@
+package com.depromeet.entity.log;
```
ddoeatlog로 바꿔주세요!

Ryuhyis commented on server-domain/src/main/java/com/depromeet/entity/like/LikeIt.java:
Likeit으로 해야한다면 id, 테이블이름 등 다 통일하는게 좋을 것 같습니다. 
아니면 깔끔하게 Heart 같은 이름도 좋아요~

Highjune replied:
@Ryuhyis Heart 로 수정하겠습니다.

Ryuhyis commented on server-domain/src/main/java/com/depromeet/entity/follow/Follow.java:
```diff
@@ -0,0 +1,26 @@
+	private Long follower_id;
+
+	@Column(nullable = false)
+	private Long following_id;
```
카멜케이스로 바꿔주세요!

Highjune replied:
앗 수정하겠습니다.

Ryuhyis reviewed (commented):
자잘한거 수정만하면 바로 머지하시죠!

thisishwan2 commented:
희수님이 잘 수정해주신것 같아요!
저도 강준님이 말씀하신것과 같은 이유로 원시타입 말고 래퍼타입을 써요! 
수정완료되면 merge 하겠습니다

Highjune committed:
[feat] 팀원들 코드 리뷰 반영
- 패키지명 log -> ddoeatlog 로 변경
- Follow 테이블에서 필드명 카멜케이스로 수정
- LikeIt 테이블명 -> Heart 로 수정

Ryuhyis reviewed (approved):
LGTM👍

Highjune committed:
[feat] 팀원들 코드 리뷰 반영
- DdoeatLog -> Review 테이블로 변경

Highjune committed:
[feat] baseEntity 추가
- createdAt, updateAt, isDeleted 추가

Highjune committed:
[fix] Store 엔티티 수정으로 인한 getter 메서드 수정
- id -> storeId
